Source/Core/InputCommon/GCAdapter.cpp

```diff
@@ -397,9 +399,10 @@ void Shutdown()
 {
   StopScanThread();
 #if defined(LIBUSB_API_VERSION) && LIBUSB_API_VERSION >= 0x01000102
-  if (s_libusb_context.IsValid() && s_libusb_hotplug_enabled)
-    libusb_hotplug_deregister_callback(s_libusb_context, s_hotplug_handle);
+  if (s_libusb_context->IsValid() && s_libusb_hotplug_enabled)
+    libusb_hotplug_deregister_callback(*s_libusb_context, s_hotplug_handle);
 #endif
+  s_libusb_context.reset();
```

Comment:
should this be moved after the next line? (`Reset()`)

Reply:
We don't directly use the libusb *context* after this point, but I'm not knowledgeable enough about libusb to say whether that would make a difference (and whether `s_handle` is still valid).

Reply:
The reported "crash" isn't a crash, it's a deadlock:
`GCAdapter::Reset()` is hung in a `join()` call for thread 0xe61f56. 
thread 0xe61f56 is in `poll()` (either `Read` or `Write` threadfunc). Read/Write wrap `libusb_interrupt_transfer`.

`LibusbUtils::Context::Impl::~Impl` will have already killed it's `EventThread` and called `libusb_interrupt_event_handler` and `libusb_exit`. So..i think that's probably the problem.

Reply:
@pizuz can you also try this change? (See my first comment)

Reply:
This produces a segfault:

[dolphin_segfault.txt](https://github.com/dolphin-emu/dolphin/files/8542159/dolphin_segfault.txt)

I'll try your new PR next.

Reply:
ok, that might have been fixed by one of the recent commits to libusb. so if you compile new PR yourself, make sure to `git submodule update` to include that part of PR

Reply:
I commented there. I tried the buildbot build of that PR.